src/EPiServer.SocialAlloy.Web/Social/Extensions/CommentExtensions.cs

```diff
@@ -25,31 +27,36 @@ public static class CommentExtensions
         static CommentExtensions()
         {
             service = ServiceLocator.Current.GetInstance<ICommentService>();
+            commentFilters = new CommentFilters();
         }
 
         /// <summary> 
         /// An extension method of the EPiServer.Core.IContent type to retrieve an page 
         /// of Comments posted for an IContent instance.  
         /// </summary> 
         /// <param name="content">A component stored in the content repository. Example: a product.</param> 
-        /// <param name="visible">The visibility filter by which to retrieve comments.</param>
+        /// <param name="visible">The visibility setting by which to retrieve comments.</param>
         /// <param name="offset">The zero-based start index of the comment to retrieve.</param> 
         /// <param name="size">The maximum number of comments to retreive.</param> 
         /// <returns> 
         /// Returns a page of comments.
         /// </returns> 
-        public static ResultPage<Comment> GetComments(this IContent content, Visibility visibile, int offset, int size)
+        public static ResultPage<Comment> GetComments(this IContent content, bool? visible, int offset, int size)
         {
+            var filters = new List<FilterExpression>();
             var targetReference = Reference.Create(
                   String.Format(resourceReferenceFormat, content.ContentGuid.ToString()));
 
-            var criteria = new Criteria<CommentFilter>
+            filters.Add(commentFilters.Parent.EqualTo(targetReference));
+
+            if (visible != null)
             {
-                Filter = new CommentFilter
-                {
-                    Parent = targetReference,
-                    Visibility = visibile
-                },
+                filters.Add(commentFilters.IsVisible.EqualTo(visible == true ? true : false));
```

Comment:
Nullable types expose a `.Value` property to make this sort of thing a little simpler.
`...IsVisible.EqualTo(visible.Value)`